org.eclipse.m2e.jdt/src/org/eclipse/m2e/jdt/internal/UnitTestSupport.java

```diff
@@ -0,0 +1,776 @@
+ * Copyright (c) 2024, 2024 pascal treilhes and others
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0.
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *   pascal treilhes - initial API and implementation
+ ********************************************************************************/
+
+package org.eclipse.m2e.jdt.internal;
+
+import java.io.File;
+import java.io.IOException;
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Method;
+import java.nio.file.Files;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.Set;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import org.eclipse.core.resources.IProject;
+import org.eclipse.core.runtime.CoreException;
+import org.eclipse.core.runtime.IProgressMonitor;
+import org.eclipse.debug.core.DebugPlugin;
+import org.eclipse.debug.core.ILaunchConfiguration;
+import org.eclipse.debug.core.ILaunchConfigurationType;
+import org.eclipse.debug.core.ILaunchConfigurationWorkingCopy;
+import org.eclipse.debug.core.ILaunchManager;
+import org.eclipse.jdt.core.IJavaProject;
+import org.eclipse.jdt.launching.IJavaLaunchConfigurationConstants;
+import org.eclipse.jdt.launching.JavaRuntime;
+
+import org.apache.maven.plugin.Mojo;
+import org.apache.maven.plugin.MojoExecution;
+import org.apache.maven.plugin.MojoExecutionException;
+import org.apache.maven.plugin.MojoFailureException;
+import org.apache.maven.project.MavenProject;
+
+import org.eclipse.m2e.core.MavenPlugin;
+import org.eclipse.m2e.core.embedder.IMaven;
+import org.eclipse.m2e.core.internal.IMavenConstants;
+import org.eclipse.m2e.core.project.IMavenProjectFacade;
+import org.eclipse.m2e.jdt.internal.launch.MavenRuntimeClasspathProvider;
+
+
+/**
+ * This class is used to support the launch configuration of JUnit and TestNG tests
+ */
+public class UnitTestSupport {
+
+  /**
+   * Feature flag to enable or disable the support
+   */
+  public static boolean FEATURE_ENABLED = true;
+
+  /**
+   * Logger
+   */
+  private static final Logger log = LoggerFactory.getLogger(UnitTestSupport.class);
+
+  /**
+   * org.apache.maven.surefire.api.testset.TestListResolver method name to check if a test should run
+   */
+  private static final String SHOULD_RUN_METHOD = "shouldRun";
+
+  /**
+   * org.apache.maven.plugin.surefire.AbstractSurefireMojo method name to get the included and excluded tests
+   */
+  private static final String GET_INCLUDED_AND_EXCLUDED_TESTS_METHOD = "getIncludedAndExcludedTests";
+
+  /**
+   * Launch configuration attribute for the main class
+   */
+  public static final String LAUNCH_CONFIG_MAIN_CLASS = "org.eclipse.jdt.launching.MAIN_TYPE";
```

Comment:
Please restrict the visibility as far as possible. And some constants are even unused.

Reply:
restricted but still kept public some constants used in test